extensions/typescript-language-features/src/languageFeatures/codeLens/baseCodeLensProvider.ts

```diff
@@ -22,6 +22,9 @@ export class ReferencesCodeLens extends vscode.CodeLens {
 }
 
 export abstract class TypeScriptBaseCodeLensProvider implements vscode.CodeLensProvider<ReferencesCodeLens> {
+	protected readonly subscriptions: vscode.Disposable[] = [];
+	protected changeEmitter = new vscode.EventEmitter<void>();
```

Comment:
This emitter should also get disposed of. You can inherit from `Disposable` too to make managing the disposables a little cleaner

Reply:
Thanks for the feedback.

> This emitter should also get disposed of. 

I derived the code from here:

https://github.com/microsoft/vscode/blob/0d552b5d26d922d68f87fde61382657fc5bc885b/extensions/npm/src/npmScriptLens.ts#L32-L35

Does that code also fail to dispose its emitter, or did I overlook how/where it does the right thing?

> You can inherit from Disposable too to make managing the disposables a little cleaner

By simply appending `, Disposable` to what `TypeScriptBaseCodeLensProvider` implements? Please help me understand what that will achieves.

Reply:
Yes it's likely missing there too

If you inherit from `Disposable`, then you don't have to implement `dispose()` yourself. Instead you can just call `this._register` with every disposable value

Reply:
I have pushed what I think you meant.